default/python/AI/fleet_orders.py

```diff
@@ -540,9 +574,6 @@ def issue_order(self):
                 fleet_id, self.ORDER_NAME, universe.getSystem(dest_id), universe.getSystem(system_id))
             fo.issueFleetMoveOrder(fleet_id, dest_id)
             print "Order issued: %s fleet: %s target: %s" % (self.ORDER_NAME, self.fleet, self.target)
-
-        if system_id == fleet.systemID:
-            if foAI.foAIstate.get_fleet_role(fleet_id) == MissionType.EXPLORATION:
-                if system_id in foAI.foAIstate.needsEmergencyExploration:
-                    foAI.foAIstate.needsEmergencyExploration.remove(system_id)
-            self.order_issued = True
+        ships_cur_health, ships_max_health = FleetUtilsAI.get_current_and_max_structure(fleet_id)
+        self.executed = (ships_cur_health == ships_max_health)
+        return True
```

Comment:
Do we rely on the fact that we end up in the target system after an OrderRepair? Otherwise, we could consider the mission finished as soon as we reach full health even if we are not at the target system (e.g. due to innate repair ability of the hull)

Reply:
It's not clear to me if you are trying to raise a concern, or make a suggestion.  As of this PR we *do* consider the mission finished as soon as we reach full health even if we are not yet in the target system, and that is intentional.  Do you have some concern about that?

Reply:
No, I wanted to suggest exactly that behavior and seemingly misunderstood the code. 

What threw me off is that we will call ```fo.issueFleetMoveOrder``` even if we are already at full health. I suppose that if we were to queue another order, it would overwrite the move order. And we would consider the next queued order because the order is marked as both issued and executed.